packages/core/crates/core-mixer/src/mixer.rs

```diff
@@ -0,0 +1,135 @@
+use std::time::Duration;
+
+use futures::stream::{Stream};
+use futures_lite::stream::StreamExt;
+use rand::Rng;
+
+use crate::future_extensions::StreamThenConcurrentExt;
+
+
+#[cfg(not(wasm))]
+use async_std::task::sleep as sleep;
+
+#[cfg(wasm)]
+use gloo_timers::future::sleep as sleep;
+
+
+const DELAY_MIN_MS: u64 = 0;    /// minimum delay in milliseconds
+const DELAY_MAX_MS: u64 = 200;  /// maximum delay in milliseconds
+
+
+#[cfg(feature = "wasm")]
+pub mod wasm {
+    use super::*;
+    use wasm_bindgen::prelude::*;
+    use wasm_bindgen_futures::stream::JsStream;
+    use wasm_bindgen::JsValue;
+    use utils_misc::async_iterable::wasm::{to_box_u8_stream,to_jsvalue_stream};
+    use js_sys::AsyncIterator;
+
+    #[wasm_bindgen]
+    pub struct AsyncIterableHelperMixer {
+        stream: Box<dyn Stream<Item = Result<Box<[u8]>, String>> + Unpin>,
+    }
+
+    pub fn new(packet_input: AsyncIterator) -> Result<AsyncIterableHelperMixer,String> {
```

Comment:
We usually annotate constructors with `#[wasm_bindgen(constructor)]`, but maybe it is not required anymore.

Reply:
Looks like it isn't necessary.